src/datasources/__tests__/grid.test.js

```diff
@@ -0,0 +1,290 @@
+
+    describe('when a cursor is requested', () => {
+      test('requests a cursor', async () => {
+        nock(/example/)
+          .defaultReplyHeaders({
+            'CMR-Hits': 84,
+            'CMR-Took': 7,
+            'CMR-Request-Id': 'abcd-1234-efgh-5678',
+            'CMR-Search-After': '["xyz", 789, 999]'
+          })
+          .post(/grids\.umm_json/)
+          .reply(200, {
+            items: [{
+              meta: {
+                'concept-id': 'GRD100000-EDSC'
+              },
+              umm: {
+                LongName: 'grid_name'
+              }
+            }]
+          })
+
+        const response = await gridDatasource({}, { headers: { 'CMR-Request-Id': 'abcd-1234-efgh-5678' } }, requestInfo)
+
+        expect(response).toEqual({
+          count: 84,
+          cursor: 'eyJ1bW0iOiJbXCJ4eXpcIiwgNzg5LCA5OTldIn0=',
+          items: [{
+            conceptId: 'GRD100000-EDSC',
+            longName: 'grid_name'
+          }]
+        })
+      })
+    })
+  })
+
+  describe('without params', () => {
+    test('returns the parsed grid results', async () => {
+      nock(/example/)
+        .defaultReplyHeaders({
+          'CMR-Hits': 84,
+          'CMR-Took': 7,
+          'CMR-Request-Id': 'abcd-1234-efgh-5678'
+        })
+        .post(/grids\.json/)
+        .reply(200, {
+          items: [{
+            concept_id: 'GRD100000-EDSC'
+          }]
+        })
+
+      const response = await gridDatasource({}, { headers: { 'CMR-Request-Id': 'abcd-1234-efgh-5678' } }, requestInfo)
+
+      expect(response).toEqual({
+        count: 84,
+        cursor: null,
+        items: [{
+          conceptId: 'GRD100000-EDSC'
+        }]
+      })
+    })
+  })
+
+  describe('with params', () => {
+    test('returns the parsed grid results', async () => {
+      nock(/example/)
+        .defaultReplyHeaders({
+          'CMR-Hits': 84,
+          'CMR-Took': 7,
+          'CMR-Request-Id': 'abcd-1234-efgh-5678'
+        })
+        .post(/grids\.json/, 'concept_id=GRD100000-EDSC')
+        .reply(200, {
+          items: [{
+            concept_id: 'GRD100000-EDSC'
+          }]
+        })
+
+      const response = await gridDatasource({ params: { concept_id: 'GRD100000-EDSC' } }, { headers: { 'CMR-Request-Id': 'abcd-1234-efgh-5678' } }, requestInfo, 'grid')
+
+      expect(response).toEqual({
+        count: 84,
+        cursor: null,
+        items: [{
+          conceptId: 'GRD100000-EDSC'
+        }]
+      })
+    })
+  })
+
+  describe('with only umm keys', () => {
+    beforeEach(() => {
+      // Overwrite default requestInfo
+      requestInfo = {
+        name: 'grids',
+        alias: 'grids',
+        args: {},
+        fieldsByTypeName: {
+          GridList: {
+            items: {
+              name: 'items',
+              alias: 'items',
+              args: {},
+              fieldsByTypeName: {
+                Grid: {
+                  longName: {
+                    name: 'grid_name',
```

Comment:
I don't think this `name` and `alias` are correct, their values should both be `longName`